jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java

```diff
@@ -71,11 +71,14 @@ public static void startHiveMonitorThread(Statement stmt,
     final ProgressBar progressBar = progressBarTemp;
     final long timeoutThreshold = Long.parseLong(
             jdbcInterpreter.getProperty("zeppelin.jdbc.hive.timeout.threshold", "" + 60 * 1000));
+    final long queryInterval = Long.parseLong(jdbcInterpreter.getProperty("zeppelin.jdbc.hive.monitor.query_interval",
+            DEFAULT_QUERY_PROGRESS_INTERVAL + ""));
     Thread thread = new Thread(() -> {
       boolean jobLaunched = false;
       long jobLastActiveTime = System.currentTimeMillis();
       while (hiveStmt.hasMoreLogs() && !Thread.interrupted()) {
         try {
+          Thread.sleep(queryInterval);
```

Comment:
Maybe we should catch the Interruptexception in a separate 'catch' to interrupt the current thread and to log a better message.

Reply:
Fixed

Reply:
For good practice we should restore the interrupted state.
Take a look at this rule:
https://rules.sonarsource.com/java/tag/multi-threading/RSPEC-2142

Maybe it makes sense to swap the ’while’ and ’try’ to reduce the amount of ’break’.

Reply:
Good idea, fixed

Reply:
`hiveStmt.isClosed())` should be part of the while condition.